bls/src/main/java/tech/pegasys/teku/bls/hashToG2/HashToCurve.java

```diff
@@ -57,28 +65,27 @@ public static boolean isInGroupG2(ECP2 point) {
    * Hashes to the G2 curve as described in the new BLS standard.
    *
    * @param message the message to be hashed. This is usually the 32 byte message digest
-   * @param cipherSuite the salt value for HKDF_Extract
+   * @param dst the domain separation tag (DST)
    * @return a point from the G2 group representing the message hash
    */
-  public static ECP2 hashToG2(Bytes message, Bytes cipherSuite) {
+  public static ECP2 hashToG2(Bytes message, Bytes dst) {
 
-    FP2Immutable u0 = hashToBase(message, (byte) 0, cipherSuite);
-    FP2Immutable u1 = hashToBase(message, (byte) 1, cipherSuite);
+    FP2Immutable[] u = hashToField(message, 2, dst);
 
-    JacobianPoint q0 = mapToCurve(u0);
-    JacobianPoint q1 = mapToCurve(u1);
+    JacobianPoint q0 = mapToCurve(u[0]);
+    JacobianPoint q1 = mapToCurve(u[1]);
 
-    JacobianPoint p = iso3(q0.add(q1));
+    JacobianPoint r = iso3(q0.add(q1));
 
     // This should never fail, and the check is non-trivial, so we use an assert
-    assert isOnCurve(p);
+    assert isOnCurve(r);
```

Comment:
nit: to be more in line with similar checks in the codebase it might be good to switch these `asserts` to `checkArgument` calls

Reply:
As per the comments in the code, the checks are very expensive and we don't want to be doing them for every hash. They are really only there as canaries if we suspect problems with the BLS stuff one day and so compile with asserts enabled.

They are also useful documentation - this is in line with Oracle's advice on the use of assertions for invariants: https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#usage-invariants

Reply:
TIL the interesting nature of assert keyword in Java. Now this makes sense.